src/File/RenameUpload.php

```diff
@@ -237,26 +107,42 @@ public function filter(mixed $value): mixed
         }
 
         // File upload via traditional SAPI
-        if (is_array($value) && isset($value['tmp_name'])) {
-            return $this->filterSapiUploadedFile($value);
+        if (
+            is_array($value)
+            && array_key_exists('tmp_name', $value)
+            && array_key_exists('name', $value)
+        ) {
+            /** @var UploadedFile $uploadedFile */
+            $uploadedFile = $value;
```

Comment:
i don't know why psalm will not detect it if i add the `@var` annotation directly to `$value`

Reply:
There's a value object you can use in `Filter\File\FileInformation`

```php
if (! FileInformation::isPossibleFile($value)) {
    return $value; // not an upload
}

$file = FileInformation::factory($value);
```

This should also negate the need for the other PSR related deps.

Psalms refusal to accept the forced `@var` is probably because of the templates